operators/tokenizer/sentencepiece_tokenizer.cc

```diff
@@ -53,6 +54,16 @@ void KernelSentencepieceTokenizer::Compute(const ortc::Tensor<std::string>& inpu
       if (add_bos) {
         content.push_back(tokenizer_.bos_id());
       }
+    } else if (xlm_roberta.has_value() && xlm_roberta) {
+      // As per HF, the first "real" token "," has position 4 in the XLMRobertaTokenizer vocab
+      // and position 3 in the SPM vocab, so we add a padding value of 1 to all IDs.
+      //std::transform(content.begin(), content.end(), content.begin(), [](int& i) { return i + 1; });
+      std::for_each(inloop.begin(), inloop.end(), [](int& n) { n++; });
+
+      // Add HF BOS (0) and EOS (2) tokens for the XLMRobertaTokenizer
+      content.push_back(xlm_bos);
```

Comment:
what's the difference of xml_bos and tokenizer_.bos_id()?

Reply:
xlm is 0 and spm is 1 - updated as per discussion yesterday